api/all/src/main/java/io/opentelemetry/api/metrics/DefaultMeter.java

```diff
@@ -51,21 +51,25 @@ static Meter getInstance() {
 
   @Override
   public LongCounterBuilder counterBuilder(String name) {
+    ValidationUtil.isValidInstrumentName(name);
```

Comment:
There doesn't seem to be a perfect name for this method but I think `checkValidInstrumentationName` returning a `boolean` is still a bit better than this seemingly-noop statement.